### PR TITLE
Publish Checksums to dotnetclichecksums storage account

### PR DESCRIPTION
**Pull request by wtgodbe**
Resolves https://github.com/dotnet/aspnetcore/issues/18792. We stopped publishing checksums some time in the 2.1 time frame - this PR turns that feature back on. Validated with https://dev.azure.com/dnceng/internal/_build/results?buildId=520228 (though that build incorrectly prefixed `RelativeBlobPath` with `Dotnet` - I've fixed that up in this PR). Still a bit of cleanup to do, but it works as intended.

@dougbu @JunTaoLuo PTAL

CC @Pilchie @mthalman

### CODE REVIEW

**Comment by wtgodbe** [comment]
Will rebase once https://github.com/dotnet/aspnetcore/pull/19036 is in

File: eng/Publishing.props [hunk 1]
```diff
@@ -27,9 +27,10 @@
         Condition=" '$(PublishInstallerBaseVersion)' == 'true' " />
     <_InstallersToPublish Include="$(ArtifactsDir)\installers\**\*.wixlib" UploadPathSegment="Runtime" />
     <_InstallersToPublish Include="$(ArtifactsDir)\installers\**\*.zip" UploadPathSegment="Runtime" />
+    <_ChecksumsToPublish Include="$(ArtifactsDir)\**\*$(ChecksumExtension)" />
```
Comment (dougbu) [inline eng/Publishing.props]: Can't see how use of `$(ChecksumExtension)` will work unless Arcade defines it in the post-built. See https://github.com/dotnet/aspnetcore/blob/1897a5d816695fdb10a918e80236507b38de5186/eng/Publishing.props#L11

**Review by dougbu — COMMENTED** [review]
Don't we need checksums for Linux and macOS installers & bundles?

**Comment by dougbu** [comment]
Please do a trial run on internal queues. Big piece of this won't work except in a "real" build.

**Comment by Pilchie** [comment]
This looks good to me.  We need something for 2.1.x as well right?

**Comment by dougbu** [comment]
@wtgodbe any internal build or answers to questions above will be much appreciated 😈

File: eng/Publishing.props (hunk 1 above)
Comment (wtgodbe) [reply eng/Publishing.props]: Whoops, didn't realize that was the case. I added that commit as a cleanup item after I got a good internal build - will fix

**Comment by wtgodbe** [comment]
> Don't we need checksums for Linux and macOS installers & bundles?

That wasn't an original requirement, just the runtime archives & Windows installer executables - https://github.com/dotnet/aspnetcore/issues/18792#issuecomment-583117032. I did a catch-all in the Windows installer folder because it seemed cleaner than picking-and-choosing the specific output Items we wanted, and more SHA512 files doesn't hurt (e.g. runtime & core-sdk do checksums for their installers). That being said, there's no good reason _not_ to do checksums for those bundles, so I can add them.

> Please do a trial run on internal queues. Big piece of this won't work except in a "real" build.

I linked the validation build in the PR description: https://dev.azure.com/dnceng/internal/_build/results?buildId=520228. I'll run another one after applying your feedback.

> This looks good to me. We need something for 2.1.x as well right?

Yep, I'll look into what's needed for that tomorrow. It may be pretty different from what I'm doing here

**Comment by wtgodbe** [comment]
New build: https://dev.azure.com/dnceng/internal/_build/results?buildId=521702

I realized this _does_ do checksums for the OSX/Linux bundles (the tar.gz archives) - it just doesn't do so for the debs/rpms

**Comment by dougbu** [comment]
> it just doesn't do so for the debs/rpms

As long as those aren't needed, sounds good. /cc @mthalman @MichaelSimons

**Review by dougbu — APPROVED** [review]

**Comment by mthalman** [comment]
> > it just doesn't do so for the debs/rpms
> 
> As long as those aren't needed, sounds good. /cc @mthalman Matt Thalman FTE @MichaelSimons Michael Simons (.NET) FTE

Yes, that's fine.

**Comment by dougbu** [comment]
@wtgodbe please look into the incredibly high number of warnings in your internal build. What gives?

**Comment by wtgodbe** [comment]
@dougbu those are mostly the Nuget warnings about stable packages w/ prerelease dependencies. They appear to have gone away in the new build, which is rebased on top of your change that fixed that problem.

**Comment by wtgodbe** [comment]
New internal build looks good, and I've confirmed everything got published to dotnetclichecksums: 
Checksums

@dougbu @mthalman any last concerns?

**Review by mthalman — APPROVED** [review]

**Comment by dougbu** [comment]
No issues except for the new conflict in Publishing.props

**Pull request merged**